### PR TITLE
IDEM-11

### PR DESCRIPTION
**Pull request by pondpiu**
#11 

- add import private key from file button
- check whether content start with and end with `KEY BLOCK` text
- show error alert when error is thrown
- common alert title enum

### CODE REVIEW

File: app/src/navigation/screens/PGPScreen.tsx [hunk 1]
```diff
@@ -5,14 +5,34 @@ import {
...
 
 import { Button } from "../../components";
 import BottomNavBarSpacer from "../../components/BottomNavBarSpacer";
 import { useClaimValue } from "../../context/ClaimsStore";
 import usePgp from "../../hooks/usePpg";
 
+const importPrivateKeyFileFromDevice = async () => {
+  const res = await DocumentPicker.getDocumentAsync({
+    type: ["application/pgp-signature", "text/*"] // .asc and .key
+  });
```
Comment (bitcoinbrisbane) [inline app/src/navigation/screens/PGPScreen.tsx:22]: Should that be ===

**Review by bitcoinbrisbane — COMMENTED** [review]
One minor thing the check, otherwise good work!

**Review by JasminFarhi — APPROVED** [review]

File: app/src/navigation/screens/PGPScreen.tsx [hunk 2]
```diff
@@ -26,6 +46,23 @@ const PGPScreen: React.FC = () => {
...
 
+  const importPrivateKeyFromDevice = async () => {
+    try {
+      const content = await importPrivateKeyFileFromDevice();
+      if (!content) return;
+      if (!isPrivateKey(content)) throw new Error("Not a private key");
+      createPublicKey(content);
+    } catch (error: any) {
+      Alert.alert(
+        "Error",
```
Comment (bitcoinbrisbane) [inline app/src/navigation/screens/PGPScreen.tsx:57]: As discussed, lets make a constants or enum for these...

File: app/src/navigation/screens/PGPScreen.tsx [hunk 3]
```diff
@@ -26,6 +46,23 @@ const PGPScreen: React.FC = () => {
...
+        `Fail to extract private key from file \n> ${
```
Comment (bitcoinbrisbane) [inline app/src/navigation/screens/PGPScreen.tsx:58]:
```suggestion
        `Failed to extract the private key from file \n> ${
```

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Few things

File: app/src/hooks/usePpg.ts [hunk 4]
```diff
@@ -62,11 +63,11 @@ const usePgp = (): Hooks => {
...
       await pgpLocalStorage.save(pgp);
-      Alert.alert("Success!", "Your PGP key has been saved");
-    } catch (error) {
+      Alert.alert(AlertTitle.Success, "Your PGP key has been saved");
+    } catch (error: any) {
       Alert.alert(
-        "UH-OH",
-        "There was a problem generating your public Key. Please try again."
+        AlertTitle.Error,
+        `There was a problem generating your public Key.\n > ${error.message}`
```
Comment (bitcoinbrisbane) [inline app/src/hooks/usePpg.ts:70]:
```suggestion
        `There was a problem generating your public key.\n > ${error.message}`
```

**Review by bitcoinbrisbane — APPROVED** [review]
Good work!  Good gardening too.

**Review by JasminFarhi — APPROVED** [review]

**Pull request merged**